### ISSUE TITLE
EmptyLinesAroundArguments vs. anonymous modules

### ISSUE THREAD

**Issue by palkan** [bbatsov/rubocop#5219]
## Example file

```ruby
# frozen_string_literal: true

module Test # :nodoc:
  def something(a, b)
    a && b
  end

  def anything(a, b)
    a || b
  end
end

Test.prepend(Module.new do
  def something(*)
    true
  end

  def anything(*)
    false
  end
end)
```

## Expected behavior

Not sure about what should be expected. I mean, I'm not sure whether using anonymous modules this way is considered a bad practice. If not then we should allow blank lines between methods definitions.

Another problem here is the output:

```
spaces_test.rb:5:1: C: Layout/EmptyLinesAroundArguments: Empty line detected around arguments.
    a && b
^
```

It's definitely broken.

## RuboCop version

Latest master ([` a28c9b22`](https://github.com/bbatsov/rubocop/tree/a28c9b22f7d59f7b8a47fe71cb83d44c9c527743)) on ruby 2.4.2p198 (2017-09-14 revision 59899) [x86_64-darwin16].

/cc @garettarrowood

**Comment by garettarrowood** [comment]
This issue makes sense. Right now `EmptyLinesAroundArguments` does not consider the type of argument it receives. Adding a `return if node.arguments.any?(&:block_type?)` guard clause makes your example pass. But give we a little longer to make sure the cop still registers when an empty line comes somewhere before a block arg.
